src/programs/ectrans-benchmark.F90

```diff
@@ -1071,6 +1084,10 @@ subroutine get_command_line_arguments(nsmax, cgrid, iters, nfld, nlev, lvordiv,
   character(len=128) :: carg          ! Storage variable for command line arguments
   integer            :: iarg = 1      ! Argument index
 
+#ifdef USE_GPU
+  !$acc init
+#endif
```

Comment:
Why can this not be done within setup_trans0 ?
OpenACC should be implementation detail

Reply:
According to the [commit log](https://github.com/ecmwf-ifs/ectrans/commit/a2ef899376a29bf6f4263d79f6b7b5baff811eeb) this needs to be done before `mpl_init`. @marsdeno is that really the case?

Reply:
Yes, I was told that MPI_INIT benefits from knowing OpenACC/GPUs will be used, in order to set up the right comms protocols in UCX layer.

Reply:
bummer. If one is using cuda or hip directly without OpenACC, how would you do this then?

Reply:
Where is `USE_GPU` defined anyway? There's only one reference to it:

```
ac6-102:/ec/res4/hpcperm/nash/ectrans $ ack USE_GPU
src/programs/ectrans-benchmark.F90
1090:#ifdef USE_GPU

src/trans/gpu/internal/trgtol_mod.F90
569:#ifdef USE_GPU_AWARE_MPI
691:#ifdef USE_GPU_AWARE_MPI

src/trans/gpu/internal/trmtol_mod.F90
168:#ifdef USE_GPU_AWARE_MPI
181:#ifdef USE_GPU_AWARE_MPI

src/trans/gpu/internal/trltog_mod.F90
695:#ifdef USE_GPU_AWARE_MPI
730:#ifdef USE_GPU_AWARE_MPI

src/trans/gpu/internal/trltom_mod.F90
174:#ifdef USE_GPU_AWARE_MPI
187:#ifdef USE_GPU_AWARE_MPI

src/trans/gpu/external/setup_trans0.F90
205:#ifdef USE_GPU_AWARE_MPI

src/trans/gpu/CMakeLists.txt
93:                               $<${HAVE_GPU_AWARE_MPI}:USE_GPU_AWARE_MPI>
```

Reply:
Should it be `#ifdef ACCGPU`? (And if so, does this mean this code wasn't actually called before??)

Reply:
Good catch Sam, this (USE_GPU) is something I have in my testing on AC that got lost in the rgo-develop preparation.

Reply:
> bummer. If one is using cuda or hip directly without OpenACC, how would you do this then?

Good question, I'll try and find out

Reply:
Is there any documentation on this `acc init` and `MPI_Init`? Which MPI implementation?
Is there perhaps an environment variable as well?
It seems to have worked seemingly without as well.

Reply:
In general, it should also work without, but the default behaviour is much better if you do acc init before MPI_Init, because then MPI knows about which GPUs you are using, and so on. You can always tweak UCX variables to achieve the same, but I highly recommend to do acc init before MPI_init.

Reply:
I think then we have to keep it for now.